src/BlockTypes/ProductRating.php

```diff
@@ -113,14 +113,22 @@ protected function render( $attributes, $content, $block ) {
 			$styles_and_classes            = StyleAttributesUtils::get_classes_and_styles_by_attributes( $attributes, array( 'font_size', 'margin', 'text_color' ) );
 			$text_align_styles_and_classes = StyleAttributesUtils::get_text_align_class_and_style( $attributes );
 
+			$average_rating = $product->get_average_rating();
+			$rating_count   = $product->get_rating_count();
+			$reviews_link   = $product->get_permalink() . '#reviews';
+			$link_label     = __( 'Add review', 'woo-gutenberg-products-block' );
+			$link_html      = '<a class="wc-block-components-product-rating__link" href="' . $reviews_link . '">' . esc_attr( $link_label ) . '</a>';
+
+			$content = $rating_count ? wc_get_rating_html( $average_rating ) : $link_html;
+
 			return sprintf(
 				'<div class="wc-block-components-product-rating wc-block-grid__product-rating %1$s %2$s" style="%3$s">
 					%4$s
 				</div>',
 				esc_attr( $text_align_styles_and_classes['class'] ?? '' ),
 				esc_attr( $styles_and_classes['classes'] ),
 				esc_attr( $styles_and_classes['styles'] ?? '' ),
-				wc_get_rating_html( $product->get_average_rating() )
+				$content
```

Comment:
Since we are already adding a filter to modify WooCommerce's default rating HTML here on this Class (see the `filter_rating_html` method on `L78`), my recommendation is to consolidate these updates on that method instead. E.g.:

```php
	public function filter_rating_html( $html, $rating, $count ) {
		$product_permalink = get_permalink();
		if ( 0 < $rating || false === $product_permalink ) {
			/* translators: %s: rating */
			$label   = sprintf( __( 'Rated %s out of 5', 'woo-gutenberg-products-block' ), $rating );
			$content = '<div class="wc-block-components-product-rating__stars wc-block-grid__product-rating__stars" role="img" aria-label="' . esc_attr( $label ) . '">' . wc_get_star_rating_html( $rating, $count ) . '</div>';
		} else {
			$product_review_url = esc_url( $product_permalink . '#reviews' );
			$content            = '<a class="wc-block-components-product-rating__link" href="' . $product_review_url . '">' . __( 'Add review', 'woo-gutenberg-products-block' ) . '</a>';
		}

		return $content;
	}
```

Reply:
First of all: thank you for this, that was my first idea, but I realised I have no clue where to get the `product_permalink` from. I didn't know I can simply call `get_permalink()` and that will work in a context of a particular product, so... Today I Learnt, thank you! 🎉 
 
But once I tried that I found a bad consequences. This filter influences not only Products block, but also Classic Template. I'm not sure why and where's the mechanism responsible for that, but when another anchor is put inside (which "Add review" is), the Rating and a Price are pushed out of the product anchor. That brings inconsistency. Check the image out:
- on the left: product which is wrapped in anchor HTML tag. Everything has a green color, since that's how I configured links
- on the right: "Add review" is a link and it's been pushed out of product anchor. Price is red, which is a color of regular text.

<img width="508" alt="image" src="https://user-images.githubusercontent.com/20098064/208447270-97b99bb1-a02a-4f6d-8086-5a0657156834.png">

Given the above I think I would go back to the previous solution. What do you think?

Reply:
> This filter influences not only Products block, but also Classic Template.

Out of curiosity: why would we want to display the "add review" link only on blocks?

> I'm not sure why and where's the mechanism responsible for that, but when another anchor is put inside (which "Add review" is), the Rating and a Price are pushed out of the product anchor. That brings inconsistency.

I'm probably missing a piece of information here: do you mind clarifying this a bit more? Isn't this a consequence of what is being discussed [over here](https://github.com/woocommerce/woocommerce-blocks/pull/7929#issuecomment-1357748183) (stars (which aren't links) follow the text color, while "Add review" follow the link color.)? I've compared both implementations side by side (consolidating everything on `filter_rating_html` versus keeping the changes on `render`), and I do not see any differences in terms of the structure of the divs or HTML anchors.

> Given the above I think I would go back to the previous solution. What do you think?

No strong preferences on my end, although filtering the rating_html and then modifying the filtered result right after, on the same class, looks a bit redundant/confusing.

Reply:
Thanks @nefeline for response!

> Out of curiosity: why would we want to display the "add review" link only on blocks?

It's not like we want the link only on blocks. Actually it's a good step to cover Classic Template as well, but there's no high demand for that.

> I'm probably missing a piece of information here: do you mind clarifying this a bit more? 

Sure thing! Apologies it wasn't clear enough.

On the screen below you can see the that when there are stars (green), they are rendered within the product anchor (red) and the same goes for price. Product anchor wraps the whole content:
<img width="2202" alt="Screen Shot 2022-12-22 at 18 21 58 PM" src="https://user-images.githubusercontent.com/20098064/209191674-7ba2f2ab-7207-423b-96e4-9ccf316592e3.png">

However, when "Add Review" link is rendered, the Product anchor gets closed (I think browser may sanitize that) to avoid nested anchors (which is forbidden). The consequence is that "Add Review" link (green) and price are already outside of Product anchor (red). That changes the clicking area for product and causes potential UI differences if a user changed the global styles. It cannot be easily fixed as Classic Template "delegates" single product rendering to WC Core.
<img width="2200" alt="Screen Shot 2022-12-22 at 18 21 31 PM" src="https://user-images.githubusercontent.com/20098064/209191668-12f668b2-3258-4148-a319-a837c0b59858.png">

Having said that, I think we should skip the Classic Template. Let me know if you have some idea how to overcome this, because I have to agree with your words:

> filtering the rating_html and then modifying the filtered result right after, on the same class, looks a bit redundant/confusing.

Reply:
Thanks @nefeline for the effort you took to show me why the above case is an edge case. It only happens when both: Products block and Classic Template are used on the same page (which I did to make the testing easier and didn't understand it influences the result).

When Classic Template is used alone on the page it stays untouched. It's very unlikely a merchant would use both those components on the same page, hence I applied the change as per your recommendation.

Having said that I think PR is ready for a final check! 🙌

Reply:
Thanks for the update here @kmanijak ! This looks good now that we have removed the redundancy of modifying an already-filtered result.

> It only happens when both: Products block and Classic Template are used on the same page

Not at all related to this PR in particular since the filter is pre-existing on the `ProductRating` class, but to prevent this behavior altogether and keep the Product Rating block customizations isolated, we would have to stop relying on the `'woocommerce_product_get_rating_html'` filter and work ourselves on a parallel implementation instead.

> It's very unlikely a merchant would use both those components on the same page

Yes, I agree: and even if they do, it still renders great and works as expected, except, of course, the custom block styles might not be applied on the Classic Template for obvious reasons :)